store/sqlstore/role_supplier.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sqlstore
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/store"
+)
+
+type Role struct {
+	Id            string
+	Name          string
+	DisplayName   string
+	Description   string
+	Permissions   string
+	SchemeManaged bool
+}
+
+func FromRole(role *model.Role) *Role {
+	return &Role{
+		Id:            role.Id,
+		Name:          role.Name,
+		DisplayName:   role.DisplayName,
+		Description:   role.Description,
+		Permissions:   strings.Join(role.Permissions, " "),
+		SchemeManaged: role.SchemeManaged,
+	}
+}
+
+func (role Role) ToRole() *model.Role {
+	return &model.Role{
+		Id:            role.Id,
+		Name:          role.Name,
+		DisplayName:   role.DisplayName,
+		Description:   role.Description,
+		Permissions:   strings.Fields(role.Permissions),
+		SchemeManaged: role.SchemeManaged,
+	}
+}
+
+func initSqlSupplierRoles(sqlStore SqlStore) {
+	for _, db := range sqlStore.GetAllConns() {
+		table := db.AddTableWithName(Role{}, "Roles").SetKeys(false, "Id")
+		table.ColMap("Name").SetMaxSize(64).SetUnique(true)
+		table.ColMap("DisplayName").SetMaxSize(128)
+		table.ColMap("Description").SetMaxSize(1024)
+		table.ColMap("Permissions").SetMaxSize(4096)
```

Comment:
All the current permissions concatenated have 934 characters (spaces included). Have sense 4096 as field size?

Reply:
It definitely needs to be a lot bigger than all the current permissions, because we're sure to add new ones in the future, and don't want to be frequently resizing this column. I'm actually wondering whether it should be *even bigger* than 4096. @crspeller what do you think?

Reply:
Actually, it doesn't matters, gorp use Text for any field greater than 256 characters in mysql, and in postgresql the underneath data type will be the same for a varchar(4096) or a text, so, I'm ok with leaving it with 4096. We can try to use a Text field for both anyway, but the current solution, works for me :)